### PR TITLE
Added filter form to article list.

### PR DESCRIPTION
**Pull request by n1k0**
This is now ready for review.

### CODE REVIEW

**Comment by n1k0** [comment]
@leplatrem r=?

File: src/js/components/ArticleList.js [hunk 1]
```diff
@@ -60,6 +60,59 @@ var PanelTitle = React.createClass({
...
+      {label: "Default", value: "0"},
+      {label: "Archived", value: "1"},
+      {label: "Deleted", value: "2"},
+    ]
+  },
+
+  filterClickHandler: function(name) {
+    return function(value) {
+      return function() {
+        ArticleActions.list({[name]: value});
```
Comment (n1k0) [inline src/js/components/ArticleList.js]: Side note, even if 2015 is a bit late, I'm so happy we can do this now.

Comment (leplatrem) [inline src/js/components/ArticleList.js]: Yes I agree, this is handful :)

But this will end up looking like perl code  #trolldi

File: src/js/components/App.js [hunk 2]
```diff
@@ -42,9 +43,8 @@ export default React.createClass({
           <div className="col-md-9">
             <ArticleForm show={this.state.edit} current={this.state.current} />
             {this.state.current ?
-              <Viewer title={this.state.current.title}
-                      url={this.state.current.url}
-                      contents={this.state.currentContents} /> : null}
+              <Viewer contents={this.state.currentContents}
+                      {...this.state.current} /> : null}
```
Comment (leplatrem) [inline src/js/components/App.js:47]: 2cents: DRY but hard to follow somehow :)

File: src/js/components/ArticleEntry.js [hunk 3]
```diff
@@ -42,9 +72,13 @@ export default React.createClass({
...
              role="group" aria-label="Actions">
-          <Button type="info" size="xs" onClick={this.handleEditClick} icon="pencil" />
-          <Button type="danger" size="xs" onClick={this.handleDeleteClick} icon="trash"
-                  disabled={this.props.status === 2} />
+          <div className="btn-group">
+            <ArchiveButton onClick={this.callAction("archive")} />
+            <MarkAsReadButton onClick={this.callAction("markAsRead")} />
+            <EditButton onClick={this.callAction("edit")} />
+            <DeleteButton onClick={this.callAction("delete", {confirm: true})}
+                          disabled={this.props.status === 2} />
```
Comment (leplatrem) [inline src/js/components/ArticleEntry.js]: We probably would like to have a constant here... (this is exactly why @Natim and @ametaireau preferred to have `deleted=true` instead of Pocket like `status=2`, as proposed by @rnewman)

File: src/js/flux.js [hunk 4]
```diff
@@ -246,7 +249,32 @@ export var ArticleStore = DocBrown.createStore({
   },
 
   markAsRead: function(article) {
-    // XXX todo
+    return this.api.updateArticle({
+      id: article.id,
+      unread: false,
+      marked_read_by: "readinglist-client",
```
Comment (leplatrem) [inline src/js/flux.js]: constant ? env ?

File: src/js/flux.js [hunk 5]
```diff
@@ -246,7 +249,32 @@ export var ArticleStore = DocBrown.createStore({
...
+  markAsReadSuccess: function() {
+    ArticleActions.list();
+  },
+
+  markAsReadError: function(err) {
+    this.setError(err, "markAsRead");
+  },
+
+  archive: function(article) {
+    return this.api.updateArticle({id: article.id, status: "1"});
```
Comment (leplatrem) [inline src/js/flux.js]: I see several arbitrary "1" here and there :)

File: src/js/components/ArticleList.js [hunk 6]
```diff
@@ -60,9 +60,64 @@ var PanelTitle = React.createClass({
...
+var Filters = React.createClass({
+  statics: {
+    unreadChoices: [
+      {label: "Read", value: false},
+      {label: "Unread", value: true}
+    ],
+    statusChoices: [
+      {label: "Default", value: "0"},
+      {label: "Archived", value: "1"},
+      {label: "Deleted", value: "2"},
```
Comment (leplatrem) [inline src/js/components/ArticleList.js]: I don't think deleted articles are meant to be displayed in the UI. When fetching them, you should remove them from your client storage and forget them forever.

Otherwise it's more like a trash, and you need an additionnal deletion notion

**Comment by leplatrem** [comment]
Now that I see the client code, I truely believe we should have chosen `"archived":true` and `"deleted": true` :) We can reiterate on this next week

**Comment by leplatrem** [comment]
r+

**Comment by rnewman** [comment]
We can still make that switch!
